gpytorch/lazy/lazy_tensor.py

```diff
@@ -1335,22 +1338,45 @@ def root_decomposition(self):
                 "Got a {} of size {}.".format(self.__class__.__name__, self.size())
             )
 
-        if (
-            self.size(-1) <= settings.max_cholesky_size.value()
-            or settings.fast_computations.covar_root_decomposition.off()
-        ):
+        if method is None:
+            if (
+                self.size(-1) <= settings.max_cholesky_size.value()
+                or settings.fast_computations.covar_root_decomposition.off()
+            ):
+                method = "cholesky"
+            else:
+                method = "lanczos"
+
+        if method == "cholesky":
             try:
                 res = self.cholesky()
                 return CholLazyTensor(res)
-
             except RuntimeError as e:
                 warnings.warn(
-                    "Runtime Error when computing Cholesky decomposition: {}. Using RootDecomposition.".format(e),
+                    f"Runtime Error when computing Cholesky decomposition: {e}. Using RootDecomposition.".format(e),
                     NumericalWarning,
                 )
+                method = "symeig"
+
+        if method == "pivoted_cholesky":
+            return RootLazyTensor(pivoted_cholesky(self.evaluate(), max_iter=100))
+
+        if method == "symeig":
+            evals, evecs = self.symeig(eigenvectors=True)
+            # TODO: only use non-zero evals (req. dealing w/ batches...)
+            F = evecs * evals.clamp(0.0).sqrt().unsqueeze(-2)
+            return RootLazyTensor(F)
+
+        if method == "svd":
```

Comment:
When would we ever use SVD for this task? The only cases where it'd be applicable are cases where symeig would apply.

Reply:
Small singular matrices on the GPU, since symeig I think is still terribly slow there. But yes, it's true that if that wasn't the case `symeig` would be preferable. Can't hurt to provide the option I guess?